Source/Core/DolphinWX/VideoConfigDiag.cpp

```diff
@@ -974,6 +972,51 @@ VideoConfigDiag::VideoConfigDiag(wxWindow* parent, const std::string& title)
   UpdateWindowUI();
 }
 
+void VideoConfigDiag::Event_Backend(wxCommandEvent& ev)
+{
+  auto& new_backend = g_available_video_backends[ev.GetInt()];
+
+  if (g_video_backend != new_backend.get())
+  {
+    bool do_switch = !Core::IsRunning();
+    if (new_backend->GetName() == "Software Renderer")
+    {
+      do_switch =
+          (wxYES ==
+           wxMessageBox(_("Software rendering is an order of magnitude slower than using the "
+                          "other backends.\nIt's only useful for debugging purposes.\nDo you "
+                          "really want to enable software rendering? If unsure, select 'No'."),
+                        _("Warning"), wxYES_NO | wxNO_DEFAULT | wxICON_EXCLAMATION, this));
+    }
+
+    if (do_switch)
+    {
+      // TODO: Only reopen the dialog if the software backend is
+      // selected (make sure to reinitialize backend info)
+      // reopen the dialog
+      Close();
+
+      g_video_backend = new_backend.get();
+      SConfig::GetInstance().m_strVideoBackend = g_video_backend->GetName();
+
+      g_video_backend->ShowConfig(GetParent());
+    }
+    else
+    {
+      // Select current backend again
+      choice_backend->SetStringSelection(StrToWxStr(g_video_backend->GetName()));
+    }
+  }
+
+  ev.Skip();
+}
+
+void VideoConfigDiag::Event_Adapter(wxCommandEvent& ev)
+{
+  // TODO
```

Comment:
This TODO appears to be 7 years old, do you think it is still accurate (not to mention we don't really know why its a TODO, since neobrain didn't specify in the comment)? If I understood `wxEvent::Skip` correctly, it just tells the event system to continue looking at other handlers...other than not having another one (at least not any obvious ones?), couldn't we just drop the event instead?